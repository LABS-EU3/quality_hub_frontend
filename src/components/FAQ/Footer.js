import React from 'react';
import styled from 'styled-components';

import facebookIcon from '../../img/fb.png';
import githubIcon from '../../img/github-1.png';
import instagramIcon from '../../img/instagram-logo-1.png';
import linkedInIcon from '../../img/linkedIn.png';
import twitterIcon from '../../img/twitter-1.png';
import logo from '../../img/firelogo.png';

// styled-components

// Footer

const NavTitleContainer = styled.div`
  font-family: 'Ubuntu';
  font-style: normal;
  font-weight: normal;
  color: grey;
  font-size: 18px;
`;

const FooterConatiner = styled.div`
  height: 50vh;
  width: 100%;
  background: #292d38;
  p {
    font-family: Ubuntu;
    font-style: normal;
    font-weight: normal;
    font-size: 12px;
    line-height: 21px;
    text-align: center;
    color: #ffffff;
  }
`;

const FooterTopSection = styled.div`
  display: flex;
  justify-content: space-evenly;
`;

const LogoFooter = styled.div`
  margin-top: 1.5%;
  margin-left: 10%;
  height: 10em;
  width: 5em;
  background-image: url(${logo});
  background-repeat: no-repeat;
`;

const LogoTitleContainerFooter = styled.div`
  margin-top: 5%;
  width: 30%;
  display: flex;
  justify-content: flex-start;
`;

const FooterTextStyle = styled.div`
  margin-top: 2%;
  h3 {
    font-family: Ubuntu;
    font-style: normal;
    font-weight: bold;
    font-size: 12px;
    line-height: 21px;
    display: flex;
    align-items: center;
    text-align: center;
    color: #ffffff;
  }
  p {
    font-family: Ubuntu;
    font-style: normal;
    font-weight: normal;
    font-size: 12px;
    line-height: 21px;
    display: flex;
    align-items: center;
    text-align: center;
    color: #ffffff;
  }
`;

const FooterBottomSection = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Icons = styled.div`
  margin-right: 1.5%;
  height: 1.6em;
  img {
    max-height: 100%;
  }
`;

export const Footer = () => {
  return (
    <FooterConatiner>
      <FooterTopSection>
        <LogoTitleContainerFooter>
          <LogoFooter />
          <NavTitleContainer>
            <h1>DevCoach</h1>
          </NavTitleContainer>
        </LogoTitleContainerFooter>
        <FooterTextStyle>
          <h3>SUPPORT</h3>
          <p>How it works</p>
        </FooterTextStyle>
        <FooterTextStyle>
          <h3>PRODUCT</h3>
          <p>Product feedback</p>
          <p>About the team</p>
        </FooterTextStyle>
        <FooterTextStyle>
          <h3>HELP</h3>
          <p>FAQ</p>
          <p>Contact us</p>
          <p>Career</p>
        </FooterTextStyle>
        <FooterTextStyle>
          <h3>LEGAL</h3>
          <p>Privacy Policy</p>
          <p>Terms and Conditions</p>
          <p>Manage Cookies</p>
        </FooterTextStyle>
      </FooterTopSection>
      <FooterBottomSection>
        <Icons>
          <img src={githubIcon} alt='social link' />
        </Icons>
        <Icons>
          <img src={facebookIcon} alt='social link' />
        </Icons>
        <Icons>
          <img src={instagramIcon} alt='social link' />
        </Icons>
        <Icons>
          <img src={linkedInIcon} alt='social link' />
        </Icons>
        <Icons>
          <img src={twitterIcon} alt='social link' />
        </Icons>
      </FooterBottomSection>
      <p>
        {' '}
        hello@dev-coach.com - copyright Â© Devcoach 2019. All rights
        reserved.
      </p>
    </FooterConatiner>
  );
};

export default Footer;
