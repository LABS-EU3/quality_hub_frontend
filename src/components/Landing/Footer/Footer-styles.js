import styled from 'styled-components';
import devices from '../../devices';

export const FooterContainer = styled.div`
  width: 100%;
  background: #292d38;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;

  .footer-top {
    display: flex;
    flex-direction: column;
    margin: 2rem 1rem 4rem 1rem;
    justify-content: center;

    .cta-message {
      margin-bottom: 1.6rem;

      h2 {
        color: #efefef;
        font-size: 2rem;
        font-weight: bold;
        text-align: center;
        line-height: 1.4;
      }
    }
  }

  .footer-bottom {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
    align-items: center;
    padding-bottom: 1.6rem;
    color: #efefef;

    @media ${devices.tablet} {
      flex-direction: column;
    }

    .footer-content {
      width: 15rem;
      opacity: 0.3;

      @media ${devices.tablet} {
        padding: 0.5rem 0;
        text-align: center;
      }
    }
  }

  .footer-icons {
    display: flex;
    justify-content: space-between;

    img {
      width: 25px;
      height: 25px;
    }
  }

  .footer-tribute {
    text-align: right;
  }
`;
