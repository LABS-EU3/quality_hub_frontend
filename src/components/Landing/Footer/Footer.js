import React from 'react';
import { StyledButton, buttonTheme } from '../Landing-styles';
import { FooterContainer } from './Footer-styles';

import github from '../../../img/github-1.png';
import facebook from '../../../img/facebook.png';
import linkdin from '../../../img/linkedin.png';
import twitter from '../../../img/twitter-1.png';
import instagram from '../../../img/instagram-logo-1.png';

const Footer = () => (
  <FooterContainer className='footer-container'>
    <div className='footer-top'>
      <div className='cta-message'>
        <h2>What are you waiting for?</h2>
      </div>
      <div className='cta-button'>
        <a href='/register'>
          <StyledButton theme={buttonTheme}>Join Now</StyledButton>
        </a>
      </div>
    </div>
    <div className='footer-bottom'>
      <div className='footer-content footer-copyright'>
        <p>Copyright Â© 2020 DevCoach</p>
      </div>
      <div className='footer-content footer-icons'>
        <div className='footer-icon'>
          <img src={github} alt='github' />
        </div>
        <div className='footer-icon'>
          <img src={facebook} alt='facebook' />
        </div>
        <div className='footer-icon'>
          <img src={twitter} alt='twitter' />
        </div>
        <div className='footer-icon'>
          <img src={linkdin} alt='linkdin' />
        </div>
        <div className='footer-icon'>
          <img src={instagram} alt='instagram' />
        </div>
      </div>
      <div className='footer-content footer-tribute'>
        <p>Designed by Labs EU3</p>
      </div>
    </div>
  </FooterContainer>
);

export default Footer;
