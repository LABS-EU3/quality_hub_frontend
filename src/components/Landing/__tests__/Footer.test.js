import React from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import * as rtl from '@testing-library/react';
import renderer from 'react-test-renderer';
import Footer from '../Footer/Footer';

describe('Footer', () => {
  it('matches snapshot ', () => {
    const tree = renderer
      .create(
        <Router>
          <Footer />
        </Router>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router>
        <Footer />
      </Router>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Register link ok', () => {
    const tree = rtl.render(
      <Router>
        <Footer />
      </Router>,
    );
    const registerLink = tree.getByTestId('registerLink');
    expect(registerLink).toBeInTheDocument();
    expect(registerLink.closest('a')).toHaveAttribute(
      'href',
      '/register',
    );
  });
});
