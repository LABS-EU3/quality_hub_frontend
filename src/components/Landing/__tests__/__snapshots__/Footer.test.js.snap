// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot  1`] = `
<div
  className="sc-bxivhb TZoTl footer-container"
>
  <div
    className="footer-top"
  >
    <div
      className="cta-message"
    >
      <h2>
        What are you waiting for?
      </h2>
    </div>
    <div
      className="cta-button"
      data-testid="cta-button"
    >
      <a
        data-testid="registerLink"
        href="/register"
        onClick={[Function]}
      >
        <button
          className="sc-htpNat bypFkS"
        >
          Join Now
        </button>
      </a>
    </div>
  </div>
  <div
    className="footer-bottom"
  >
    <div
      className="footer-content footer-copyright"
    >
      <p>
        Copyright Â© 2020 DevCoach
      </p>
    </div>
    <div
      className="footer-content footer-icons"
    >
      <div
        className="footer-icon"
      >
        <img
          alt="github"
          src="github-1.png"
        />
      </div>
      <div
        className="footer-icon"
      >
        <img
          alt="facebook"
          src="facebook.png"
        />
      </div>
      <div
        className="footer-icon"
      >
        <img
          alt="twitter"
          src="twitter-1.png"
        />
      </div>
      <div
        className="footer-icon"
      >
        <img
          alt="linkdin"
          src="linkedin.png"
        />
      </div>
      <div
        className="footer-icon"
      >
        <img
          alt="instagram"
          src="instagram-logo-1.png"
        />
      </div>
    </div>
    <div
      className="footer-content footer-tribute"
    >
      <p>
        Designed by Labs EU3
      </p>
    </div>
  </div>
</div>
`;
